### ISSUE TITLE
Alpine aarch64/arm64 images

### ISSUE THREAD

**Issue by gregorkistler** [corretto/corretto-docker#46]
Any plans yet to build and provide also Docker images for architectures other than x86? :)

```docker pull amazoncorretto:8-alpine-jre
8-alpine-jre: Pulling from library/amazoncorretto
no matching manifest for linux/arm64/v8 in the manifest list entries
```

Building manually will also fail as the custom repo `apk.corretto.aws` does not contain any aarch64 directory:
```apk update
fetch https://apk.corretto.aws/aarch64/APKINDEX.tar.gz
ERROR: https://apk.corretto.aws: Permission denied
```

Thanks guys!

**Comment by venubabu452** [comment]
Hi,
Exactly same problem here as well, we are blocked because of this.
Please try to fix this ASAP.

Thanks,
Venu

**Comment by davecurrie** [comment]
Hi. Thanks for getting in touch. We don't yet have builds for Alpine Linux aarch64. We've taken note of the request.

**Comment by siwyd** [comment]
This would be great yeah. We use Jib to build multi-platform images for amd64 and arm64. It would be nice to be able to switch to the smaller Alpine images.

**Comment by nicolabeghin** [comment]
any update on this? it's a show-stopper on AWS Graviton instances.

**Comment by tigerinus** [comment]
patiently waiting for this AWS Graviton show stopper to be resolved....

**Comment by choeflake** [comment]
> patiently waiting for this AWS Graviton show stopper to be resolved....

Same here!

**Comment by pdehlke** [comment]
Bump. No progress on this? Seems weird given the investment in Gravition.

**Comment by DennisBecker** [comment]
Any information about this? Don't understand what the problem is as alpine offers linux/arm64/v8 images and Coretto has arm64 builds available.

**Comment by wesleymooiman** [comment]
This is really blocking me i want the use the amazoncorretto:11-alpine image as base for my project which i run on aws graviton. I don't understand why there is no arm64 build alpine itself is available  as arm64....

**Comment by dankirkd** [comment]
@davecurrie Can someone provide an update - it has been over a year now.  We'd also like to play with Graviton, and M1 Macs are becoming more prevalent, but a lack of a Corretto ARM based Docker image is somewhat of an impediment that has us having to use other approaches (such as an OpenJDK/Debian image from https://hub.docker.com/r/arm64v8/openjdk/).

**Comment by pexa-abruno** [comment]
I am also facing similar issues when using `amazoncorretto:11-alpine`

Error: Could not pull image amazoncorretto:11-alpine.
Pulling image 'amazoncorretto:11-alpine' failed: no matching manifest for linux/arm64/v8 in the manifest list entries

If I use `amazoncorretto:11` it works across both Apple Intel and M1.

Looking for an amazoncorretto alpine image that works across both Apple Intel and M1.  Any advice/suggestions?

**Comment by jbg** [comment]
@pexa-abruno As far as I can tell, there are no musl/arm64 Corretto binaries, which are a prerequisite for making an image. You would have to [build Corretto from source](https://github.com/corretto/corretto-11/blob/develop/doc/building.md) for musl/arm64, or wait for Amazon to release binaries and an image.

It's bizarre that it takes so long since they already have musl/amd64 and glibc/arm64, so it's most likely just a few CI/CD changes to build the extra target...

In the meantime you could use the amd64 alpine image on M1 (with the Rosetta performance hit) by passing `--platform linux/amd64` to Docker.

**Comment by davecurrie** [comment]
It's more complicated than just a CI/CD change, there is a lot of work that goes into validating new Corretto builds and making sure we can consistently deliver quarterly updates in sync with existing builds. :)

We plan to deliver this and work should be starting soon. We will likely start with Corretto 17, since Alpine code is integrated into upstream OpenJDK 17, and work down from there to 11 and 8.

We'll keep this issue updated with progress and any preview builds. Thanks for your patience.

**Comment by dankirkd** [comment]
I'd put in a plug for starting with 11 and 8.  That's what most people are using.

**Comment by pexa-abruno** [comment]
I have found `adoptopenjdk/openjdk11:jdk11u-nightly-slim` to work across both Apple Intel and M1

**Comment by andrewbruno** [comment]
I have found `adoptopenjdk/openjdk11:jdk11u-nightly-slim` to work across both Apple Intel and M1

**Comment by jbg** [comment]
@davecurrie maybe i'm mistaken but aren't you already shipping Alpine amd64 binaries for 17, 11 and 8? So out of curiosity, why does it matter that OpenJDK 17 is already supporting Alpine? Corretto is already supporting Alpine/musl across the board. This is "just" building for a new arch, I can of course understand the need for verification though.

**Comment by lutkerd** [comment]
> I have found `adoptopenjdk/openjdk11:jdk11u-nightly-slim` to work across both Apple Intel and M1, however for production releases I would prefer to use `amazoncorretto:11-alpine`

The adoptopenjdk docker project is deprecated and shouldn't be used. The image you mentioned is over 6 months out of date and missing the last few quarters of security patches. The page says `DEPRECATED; use https://hub.docker.com/_/eclipse-temurin/`

**Comment by andrewbruno** [comment]
Thank you @lutkerd - Been using `openjdk:11` https://hub.docker.com/_/openjdk with success.

**Comment by sakshith007** [comment]
We are building multi architecture docker alpine image. And facing the same issue as discussed with installing in arm64 alpine while for amd64 it gets installed without any issue. Is corretto-jdk available now for alpine arm64 or is there any method i could achieve this . Anyone's reponse is much appreciated . Thanks

**Comment by Rudometov** [comment]
Right now there is no Alpine arm64 Corretto JDK available. However the work has been started and we plan to deliver it in the future.

**Comment by paul-murage** [comment]
@Rudometov is there a timeline for when the Alpine arm64 Corretto JDK will be available ?

**Comment by MrDienns** [comment]
Could we please get a timeline for when this will be available?

**Comment by davecurrie** [comment]
It's great to see all the interest and we are planning to deliver as soon as possible. I can't post dates but we will update this issue as soon as we have something to share.

**Comment by pdehlke** [comment]
I mean it's nice to hear from a committer here, but I hope you'll understand the frustration with "we're gonna fix this, really we are!" for going on two years now :(

**Comment by janario** [comment]
Really looking forward to this 🤞 

As I see it needs the release of such package at Alpine repos first, so I thought we could push it over something more specific https://github.com/corretto/corretto-17/issues/96

**Comment by bogdandrutu** [comment]
@davecurrie any update on this one?

**Comment by dankirkd** [comment]
@bogdandrutu do these not qualify? https://hub.docker.com/r/arm64v8/amazoncorretto/

**Comment by bchilcott** [comment]
Throwing my name in the pile as someone who would also really appreciate this 👍🏼

**Comment by felixsilberstein** [comment]
Impacted by this issue too, thanks for the work.

**Comment by dankirkd** [comment]
We've given up waiting for such a thing and found that the Zulu images at https://github.com/zulu-openjdk/zulu-openjdk/tree/master/alpine are fine alternatives to Corretto, with comparable LTS lifespans https://www.azul.com/products/azul-support-roadmap/

We're using the ones at https://github.com/zulu-openjdk/zulu-openjdk/tree/master/alpine/17.0.5-17.38.21

Definitely worth a look, and nice and small, and although we compile with Java 11, we've found that running on Java 17 provides a significant performance improvement with zero code changes necessary.

YMMV, but it is better than waiting for more than you already may have for a Corretto Alpine ARM64 image. 😃

**Comment by benty-amzn** [comment]
Corretto 8, 11, 17, and 19 are now available for Alpine Linux aarch64 as a preview release.

For this preview, you can find Dockerfiles on the branch [`alpine-aarch64-prerelease`](https://github.com/corretto/corretto-docker/tree/alpine-aarch64-prerelease). 

The preview JDKs are available through:
-  Corretto [Alpine Linux apk repo](https://docs.aws.amazon.com/corretto/latest/corretto-17-ug/generic-linux-install.html#alpine-linux-install-instruct)
- [corretto-8](https://github.com/corretto/corretto-8/releases/tag/8.362.10.1)
- [corretto-11](https://github.com/corretto/corretto-11/releases/tag/11.0.18.11.1)
- [corretto-17](https://github.com/corretto/corretto-17/releases/tag/17.0.6.10.1)
- [corretto-19](https://github.com/corretto/corretto-19/releases/tag/19.0.2.7.1)

Feedback is welcome and requested, please report any issues you encounter.

**Comment by dankirkd** [comment]
@benty-amzn Are these available from docker.io?

**Comment by benty-amzn** [comment]
Images are not currently available on docker.io. We will post images once we're out of preview.

**Comment by benty-amzn** [comment]
If no critical/blocking bugs are reported during the preview we intend to release these for GA with the Q2 patch release on 2023-04-18.

**Comment by ChrisRut** [comment]
> If no critical/blocking bugs are reported during the preview we intend to release these for GA with the Q2 patch release on 2023-04-18.

Any updates on this?  It doesn't look like this made it into the Q2 patch release on 2023-04-15

**Comment by shapeofarchitect** [comment]
@benty-amzn Thanks for the above merge. It seems docker.io still doesn't have the tag for alpine in corretto-17 and 19. Seems we still have to update the manifest ?

```
docker pull amazoncorretto:17-alpine-jdk
17-alpine-jdk: Pulling from library/amazoncorretto
no matching manifest for linux/arm64/v8 in the manifest list entries
```

**Comment by lutkerd** [comment]
We updated our end and now we have to wait for DockerHub to merge this change to consume our latest images: https://github.com/docker-library/official-images/pull/14471

Once that is merged it usually takes ~4 hours for the images to be live.

**Comment by benty-amzn** [comment]
docker-library PR is merged and these images should now be available. Please feel free to open a follow-up issue if you encounter any bugs.
